### ISSUE TITLE
[Mobile] Do not vibrate if an input is not sent

### ISSUE THREAD

**Issue by tytydraco** [libretro/RetroArch#11424]
## Description

When haptic vibration is enabled on a mobile device, the device will vibrate when any area on the overlay is pressed.

### Expected behavior

The device should only vibrate if a keyevent is sent. Therefore, a touch on an empty part of the overlay should not vibrate the device.

### Actual behavior

When any location on the overlay is pressed, **even if it is not a button**, the device vibrates.

### Steps to reproduce the bug

1. Enable Input --> Haptic Feedback / Vibration --> Vibrate on Key Press
2. Touch anywhere on the screen to feel a vibration

### Bisect Results

This issue has been around since the vibrate feature has been implemented.

### Version/Commit
You can find this information under Information/System Information on Android: https://github.com/libretro/RetroArch/commit/1e223733fc17a290d503fae5de80928e444bcd9f

- RetroArch: [1.9.0/129852f]

### Environment information

- OS: [Android 10]
- Compiler: [Clang/LLVM (8.0.7)]

**Comment by andreycout** [comment]
Was about to create an issue for this but thankfully searched and found this. Solution seems simple, should be easy to solve. Hope someone picks this up. Haptic feedback when you press a virtual button correctly is very important to train your muscle memory into touching the correct spot and make up for the lack of tactile feel on the touch screen. The way it's implemented right now is pretty pointless, it always vibrates regardless of where you touch.

**Comment by tytydraco** [comment]
Sadly it's been so long since I made this issue it's probably not happening. Moved away from RetroArch because it feels like on Android it behaves poorly for an Android app compared to the standard. It's not really their fault, it's because they are using a shared code base between systems so you can't really use the Android libraries that make it FEEL like an Android app.

**Comment by JooJooBee666** [comment]
Yep.  2025 and this sh1t is still b0rked to all hell...hard to belive this is still broken.  It makes this app REALLY suck on Android.  Such a waste.

UPDATE: Any better alternatives?  This bug still happening 5 years later has taken me over the edge.  I'm DONE with this trash.  So let's make this a thread for apps that actually work and fix this problem.

**Comment by warmenhoven** [comment]
If you have the mouse overlay turned on, which is the default, then any press not on an overlay button is a mouse press, which is an input and would provide haptic feedback. When I turn the mouse overlay off I can't reproduce this.

**Comment by gouchi** [comment]
Duplicate of #10756 but this issue got more information.

Also, [initial commit](https://github.com/libretro/RetroArch/pull/8442/files) for this feature.

**Comment by JooJooBee666** [comment]
> If you have the mouse overlay turned on, which is the default, then any press not on an overlay button is a mouse press, which is an input and would provide haptic feedback. When I turn the mouse overlay off I can't reproduce this.

Mouse support has been disabled through the user iterface option long ago.  Definitely NOT the problem here. 🫤

**Comment by warmenhoven** [comment]
Can you be very precise about exactly which mouse option you have disabled? There are several.

**Comment by JooJooBee666** [comment]
**User Interface** (with **Show Advanced Settings** already enabled) -> **Mouse Support** -> **disabled**.

**Comment by warmenhoven** [comment]
Right, that is most definitely not the "mouse overlay" option I was talking about.

settings > user interface > on-screen overlay > enable overlay lightgun, mouse, and pointer

**Comment by JooJooBee666** [comment]
I don't have an **On-screen Overlay** option under **User Interface**.

**Comment by warmenhoven** [comment]
What version are you using?

**Comment by JooJooBee666** [comment]
The latest from the Play store, so 1.9.12.

**Comment by warmenhoven** [comment]
That's your problem. The play store version is about four years old now. retroarch.com recommends other sources instead.

**Comment by JooJooBee666** [comment]
That needs de-listed then so it doesn't confuse people.  

What is the recommended way to do this for Android then?  I don't see any installation help on the main README here.

**Comment by warmenhoven** [comment]
De-listing it is problematic because it opens an opportunity for someone not associated with libretro to publish it without consent.

Sorry, I don't actually know how to install things on android without using Android Studio.

**Comment by JooJooBee666** [comment]
> De-listing it is problematic because it opens an opportunity for someone not associated with libretro to publish it without consent.
> 
> Sorry, I don't actually know how to install things on android without using Android Studio.

Could someone at least note that as obsolete somehow and maybe add guidance for how to get an updated version on the store page?  Or, and I know this is crazy, but maybe just update it?  I don't know all the reason behind this but this is frustrating and should be fixed.

I guess I'm off to find some random APK now, not ideal at all.

UPDATE: It looks like the listing on F-Droid is up to date.  Is this the official recommended install now?

**Comment by warmenhoven** [comment]
Updating it is challenging because Google requires moving to the scoped storage APIs and doing that breaks many existing installs. There has been some discussion around exactly how to approach the problem and some attempted fixes but so far all of the solutions to the mess Google created fall short.

It's not some random APK. Did you even look at https://www.retroarch.com/?page=platforms

**Comment by JooJooBee666** [comment]
I did not.  However, I installed via F-Droid.  This method is preferred as I won't need to manually download every time I need to update.  I just got it all switched around and confirmed the issue is resolved with the latest from there.  Much happier now. 👍
